analyzers/dataframe/src/Algorithms.cc

```diff
@@ -456,6 +456,81 @@ float getAxisCosTheta(const ROOT::VecOps::RVec<float> & axis,
   return result;
 }
 
+
+
+jets_TwoHemispheres::jets_TwoHemispheres(int arg_sorted, int arg_recombination) : m_sorted( arg_sorted ), m_recombination( arg_recombination ) {
+   if ( arg_recombination  != 0 ) {
+      std::cout << " ....... in jets_TwoHemispheres: only E-scheme is implemented so far. m_recombination  is set to zero. " << std::endl;
+      m_recombination = 0;
+   }
+};
+
+JetClustering::FCCAnalysesJet jets_TwoHemispheres::operator() ( const ROOT::VecOps::RVec<edm4hep::ReconstructedParticleData> & in) {
+
+ ROOT::VecOps::RVec<float> RP_e = ReconstructedParticle::get_e( in );
+ ROOT::VecOps::RVec<float> RP_px = ReconstructedParticle::get_px( in );
+ ROOT::VecOps::RVec<float> RP_py = ReconstructedParticle::get_py( in );
+ ROOT::VecOps::RVec<float> RP_pz = ReconstructedParticle::get_pz( in );
+
+ ROOT::VecOps::RVec<float> mt =  minimize_thrust("Minuit2","Migrad")(RP_px, RP_py, RP_pz) ;
+ ROOT::VecOps::RVec<float> costheta = getAxisCosTheta( mt, RP_px, RP_py, RP_pz) ;
+ 
```

Comment:
These should be inputs to the function. Maybe then the additional includes won't be needed.

Reply:
I had chosen the implementation that maximally hides the complexity to the user. But OK, I have updated the implementation along the lines you suggested. 

Here is the new usage:

            .Define("RP_px", "ReconstructedParticle::get_px(ReconstructedParticles) ")
            .Define("RP_py", "ReconstructedParticle::get_py(ReconstructedParticles) ")
            .Define("RP_pz", "ReconstructedParticle::get_pz(ReconstructedParticles) ")
            .Define("RP_e", "ReconstructedParticle::get_e(ReconstructedParticles) ")

            .Define("thrust",  'Algorithms::minimize_thrust("Minuit2","Migrad")(RP_px, RP_py, RP_pz)')
            .Define("RP_costheta",  "Algorithms::getAxisCosTheta( thrust, RP_px, RP_py, RP_pz)" )

            .Define("FCCAnalysesJets_hemispheres",  "Algorithms::jets_TwoHemispheres( 1, 0) ( RP_px, RP_py, RP_pz, RP_e, RP_costheta )")